kratos/utilities/math_utils.h

```diff
@@ -403,7 +403,31 @@ class MathUtils
             noalias(InvertedMatrix) = prod(auxInv, trans(InputMatrix));
         }
     }
-    
+
+    /**
+     * @brief This function is designed to be called when a dense linear system is needed to be solved
+     * @param rA System matrix
+     * @param rX Solution vector. it's also the initial guess for iterative linear solvers.
+     * @param rB Right hand side vector.
+     */
+    static void Solve(
+        const MatrixType& rA,
+        VectorType& rX,
+        const VectorType& rB
+        )
+    {
+        const SizeType size1 = rA.size1();
+
+        rX = rB;
+        typedef permutation_matrix<SizeType> pmatrix;
+        Matrix A(rA);
+        pmatrix pm(size1);
+        int singular = lu_factorize(A,pm);
+        lu_substitute(A, pm, rX);
+
+        KRATOS_ERROR_IF(singular == 1) << "::ERROR: Matrix is singular: " << rA << std::endl;
```

Comment:
i would put this as KRATOS_DEBUG_ERROR_IF

in any case even if you don't put the error before the lu_substitute

Reply:
OK